folio-spring-i18n/src/main/java/org/folio/spring/service/TranslationService.java

```diff
@@ -0,0 +1,347 @@
+   * @see <a href="https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Language"> MDN docs</a>
+   */
+  public TranslationMap getCurrentTranslation() {
+    return this.getBestTranslation(this.getCurrentLocales());
+  }
+
+  /**
+   * Get the best {@link TranslationMap TranslationMap} for the provides list of Locales.
+   * This will return the first one that matches on at least language or, if none match,
+   * the default TranslationMap.
+   *
+   * @param locales the ordered list of locales to consider
+   * @return the best TranslationMap
+   */
+  public TranslationMap getBestTranslation(Iterable<Locale> locales) {
+    // return the first one that is a good match
+    for (Locale locale : locales) {
+      TranslationMap correspondingMap = this.getTranslation(locale);
+      if (correspondingMap.getQuality() != TranslationMatchQuality.NO_MATCH) {
+        return correspondingMap;
+      }
+    }
+
+    return this.getDefaultTranslation();
+  }
+
+  /**
+   * Wraps the {@link TranslationMap#format TranslationMap#format} method on the current translation.
+   * Equivalent to {@code getCurrentTranslation().format(...)}.
+   *
+   * <p>Format an ICU format string (found by its key), supplying a series of named arguments as key
+   * value pairs.  For example: {@code format("Hello {name}", "name", parameterValue)}</p>
+   *
+   * @param key the key of the format string.  You likely want to use a constant from
+   *   {@link TranslationKey TranslationKey} rather than hard-coding a string.
+   * @param args pairs of keys and values to interpolate
+   * @return the formatted string
+   */
+  public String format(String key, Object... args) {
+    return this.getCurrentTranslation().format(key, args);
+  }
+
+  /**
+   * Format a list of strings into one cohesive string.  For example, in English, {@code [A B C D]}
+   * would become {@code "A, B, C, and D"}.
+   *
+   * @param list a list of items to join
+   * @return the joined string
+   */
+  public String formatList(Collection<?> list) {
+    return ListFormatter.getInstance(getCurrentLocale()).format(list);
+  }
+
+  /**
+   * Get all available translations in the classpath.
+   *
+   * @return a list of available translation files
+   * @throws IllegalStateException if the translation files cannot be found
+   */
+  public List<TranslationFile> getAvailableTranslationFiles() {
+    try {
+      Map<String, List<Resource>> localeGroups = Arrays
+        .asList(
+          resourceResolver.getResources(
+            String.format(
+              TRANSLATIONS_CLASSPATH,
+              configuration.getTranslationDirectory()
+            )
+          )
+        )
+        .stream()
+        .filter(Resource::isReadable)
+        .collect(Collectors.groupingBy(Resource::getFilename));
+
+      List<TranslationFile> files = localeGroups
+        .entrySet()
+        .stream()
+        .map(entry -> new TranslationFile(entry.getValue()))
+        .toList();
```

Comment:
No need to use the entry set if all we need is the values.
```suggestion
      List<TranslationFile> files = localeGroups
        .values()
        .map(TranslationFile::new)
        .toList();
```